ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java

```diff
@@ -79,31 +78,34 @@ public NettyServerStreamRpc(RaftPeer server, StateMachine stateMachine) {
     }
   }
 
-  private void writeTo(ByteBuf buf, DataStream stream, boolean released) {
+  private long writeTo(ByteBuf buf, DataStream stream, boolean released) {
     if (released) {
-      return;
+      return 0;
     }
     try {
       if (stream == null) {
-        return;
+        return 0;
       }
 
       final WritableByteChannel channel = stream.getWritableByteChannel();
+      long written = 0;
       for (ByteBuffer buffer : buf.nioBuffers()) {
         try {
-          channel.write(buffer);
+          written += channel.write(buffer);
         } catch (Throwable t) {
           throw new CompletionException(t);
         }
       }
+      return written;
     } finally {
       buf.release();
     }
   }
 
-  private void sendReply(DataStreamRequestByteBuf request, ChannelHandlerContext ctx) {
-    final DataStreamReply reply = new DataStreamReplyByteBuffer(
-        request.getStreamId(), request.getDataOffset(), ByteBuffer.wrap("OK".getBytes()));
+  private void sendReply(DataStreamRequestByteBuf request, long written, ChannelHandlerContext ctx) {
+    final boolean success = request.getDataLength() == written;
```

Comment:
`success ` seems useless.

Reply:
Yes, it is not useful for now.  I don't want to change to many code in this issue.  Just have filed https://issues.apache.org/jira/browse/RATIS-1098 as a followup.